src/main/java/com/example/employee/service/EmployeeService.java

```diff
@@ -17,8 +17,20 @@ public class EmployeeService {
 
     public EmployeeDTO createAndUpdateEmployee(EmployeeDTO employeeDTO){
         Employee employee = employeeMapper.dtoToEntity(employeeDTO);
+        if (employeeRepository.findEmployeeByFirstNameAndLastNameAndDepartmentAndPositionsAndEmailAndIsActiveAndIsPensionsPayer(
+                employeeDTO.getFirstName(),
+                employeeDTO.getLastName(),
+                employeeDTO.getDepartment(),
+                employeeDTO.getPositions(),
+                employeeDTO.getEmail(),
+                employeeDTO.getIsActive(),
+                employeeDTO.getIsPensionsPayer()
+        ) == null){
```

Comment:
This if check looks too big for me
We should figure out how we want to identify an Employee and check only by one unique identifier
It can be just `id` and then when we have update requests - we send them with `id` and if we want to create new employee - then we send request without `id`

Reply:
I used all the criteria because each of them individually can be duplicated. For instance the first name and last name can be the same for two employees and it may can cause some misleadings.

What if I add new identifier - "personal_id" String type value expressed by 11 number that are unique for each person?